moveit_ros/planning_interface/move_group_interface/src/wrap_python_move_group.cpp

```diff
@@ -536,14 +536,16 @@ class MoveGroupInterfaceWrapper : protected py_bindings_tools::ROScppInitializer
     }
   }
 
-  Eigen::MatrixXd getJacobianMatrixPython(bp::list& joint_values)
+  Eigen::MatrixXd getJacobianMatrixPython(bp::list& joint_values, bp::list& reference_point_position)
   {
-    std::vector<double> v = py_bindings_tools::doubleFromList(joint_values);
+    std::vector<double> joint_vals = py_bindings_tools::doubleFromList(joint_values);
+    std::vector<double> ref_point_pos_vector = py_bindings_tools::doubleFromList(reference_point_position);
+    Eigen::Vector3d ref_point_pos(ref_point_pos_vector.data());
```

Comment:
Use a `const Eigen::Map<Eigen::Vector3d>` wrapper here to avoid copying the data.

Reply:
Can you explain more? `getJacobian` takes an `Eigen::Vector3d` as input, and I'm not sure how to avoid copying the data via `py_bindings_tools::doubleFromList`

Reply:
You copy twice: First `doubleFromList` is used to create a `std::vector`. Then, this std::vector is copied over to an `Eigen::Vector3d`. This second copy can be avoided using an Eigen::Map, which is essentially a wrapper around already allocated memory.

Reply:
I missed this comment and am not at my PC anymore, but if simply replacing `Eigen::Vector3d` with `Eigen::Map<Eigen::Vector3d>` works, then I'll put it in on Monday.

Reply:
Done.